core/corehttp/logs.go

```diff
@@ -0,0 +1,42 @@
+package corehttp
+
+import (
+	"io"
+	"net/http"
+
+	core "github.com/ipfs/go-ipfs/core"
+	"github.com/ipfs/go-ipfs/thirdparty/eventlog"
+)
+
+type writeErrNotifier struct {
+	w    io.Writer
+	errs chan error
+}
+
+func newWriteErrNotifier(w io.Writer) (io.Writer, <-chan error) {
+	ch := make(chan error, 1)
+	return &writeErrNotifier{
+		w:    w,
+		errs: ch,
+	}, ch
+}
+
+func (w *writeErrNotifier) Write(b []byte) (int, error) {
+	n, err := w.w.Write(b)
+	if err != nil {
+		w.errs <- err
+	}
+	return n, err
+}
+
+func LogOption() ServeOption {
+	return func(n *core.IpfsNode, mux *http.ServeMux) (*http.ServeMux, error) {
+		mux.HandleFunc("/logs", func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(200)
+			wnf, errs := newWriteErrNotifier(w)
+			eventlog.WriterGroup.AddWriter(wnf)
+			<-errs
```

Comment:
@cryptix that was intended. I assume that the only error we will encounter is the http connection being closed. The first error encountered will drop the connection from the mirrorwriter, and then this will exit. I *could* add a default case to the send on the write notifier to ensure that misuse of the code doesnt cause a hang